pacsman/filesystem_dev_client.py

```diff
@@ -25,35 +25,48 @@
 import logging
 import os
 import shutil
+from typing import List, Optional, Dict, Iterable
 
 from pydicom import dcmread, Dataset
 from pydicom.valuerep import MultiValue
 
 from .dicom_interface import DicomInterface
-from .utils import process_and_write_png, copy_dicom_attributes
+from .utils import process_and_write_png, copy_dicom_attributes, dicom_filename
 
 logger = logging.getLogger(__name__)
 
 
 class FilesystemDicomClient(DicomInterface):
-    def __init__(self, dicom_dir, dicom_source_dir, *args, **kwargs):
+    def __init__(self, dicom_dir: str, dicom_source_dir: str, *args, **kwargs) -> None:
         """
         :param dicom_src_dir: source directory for *.dcm files
         :param dicom_dir: the DICOM output dir for image retrievals (same as other clients)
         """
         self.dicom_dir = dicom_dir
+        self.dicom_source_dir = dicom_source_dir
 
-        self.dicom_datasets = {}
+        self.dicom_datasets: Dict[str, Dataset] = {}
 
         for dicom_file in glob.glob(f'{dicom_source_dir}/*.dcm'):
-            filepath = os.path.join(dicom_source_dir, dicom_file)
-            self.dicom_datasets[filepath] = dcmread(filepath)
+            self._read_and_add_data_set(dicom_file)
 
-    def verify(self):
+    def _read_and_add_data_set(self, filename: str) -> None:
+        filepath = self._filepath(filename)
+        self._add_dataset(dcmread(filepath), filepath)
+
+    def _add_dataset(self, dataset: Dataset, filepath: str = None) -> None:
+        if filepath is None:
+            filepath = self._filepath(dicom_filename(dataset))
+        self.dicom_datasets[filepath] = dataset
```

Comment:
This only changes the "transient" state of the `FilesystemDicomClient` since it doesn't write it to disk. If you opened the client again in a new process, the dataset wouldn't be present. I'm conflicted because on one hand you might not want to be editing your test data directories. On the other, it doesn't match the `PynetdicomClient` because there (assuming the PACS accepts the dataset), it will be persisted. Any thoughts?

Reply:
Very good point. I think we should make the code non-transient. If someone wants transient, they can first make a copy of their test data and operate on the copy. There is not such an easy work around for the current implementation in the case someone wants the non-transient behavior.
I am adding this as an issue.